server/storage/gdrive.go

```diff
@@ -45,13 +46,20 @@ func NewGDriveStorage(clientJSONFilepath string, localConfigPath string, basedir
 	}
 
 	// If modifying these scopes, delete your previously saved client_secret.json.
-	config, err := google.ConfigFromJSON(b, drive.DriveScope, drive.DriveMetadataScope)
-	if err != nil {
-		return nil, err
+	var httpClient *http.Client
+
+	if strings.Contains(string(b), `"type": "service_account"`) {
+		logger.Println("GDrive: using Service Account credentials")
+		httpClient = JWTConfigFromJSON(b, logger).Client(ctx)
```

Comment:
`JWTConfigFromJSON` -> `getGDriveClientFromServiceAccount`

no need to pass `logger`, it's not used in the function

Reply:
Done with fb87a5b32f8acda48d5a7f8ec582ce4f5bdee8c8